tools/tesseratos/src/tesseratos/metrics_panel/plugin.cpp

```diff
@@ -0,0 +1,66 @@
+#include <imgui.h>
+
+#include <cubos/engine/imgui/plugin.hpp>
+#include <cubos/engine/imgui/serialization.hpp>
+
+#include <tesseratos/metrics_panel/plugin.hpp>
+
+using cubos::core::ecs::Read;
+using cubos::core::ecs::Write;
+using cubos::engine::Cubos;
+using cubos::engine::DeltaTime;
+
+using namespace tesseratos;
+
+struct Metrics
+{
+    float timeElapsed = 0.0;
+    float fps = 0.0;
+    int frameCounter = 0;
+    float secondTimer = 0.0;
+    // FPSGraph related
+    std::vector<float> fpsHistory;
+};
```

Comment:
You should place this type within an anonymous namespace, as it is private to this source file. Currently we don't do this everywhere but we should start doing so - it prevents weird linker errors if someone defines another type with the same name.

```cpp
namespace
{
    struct Metrics { ... };
}
```

The rest of the code remains the same :)